===================================================================
--- origin_platforms/android/CordovaLib/build.gradle	null
+++ platforms/android/CordovaLib/build.gradle	null
@@ -24,6 +24,7 @@

 buildscript {
     repositories {
+        google()
         maven {
             url "https://maven.google.com"
         }
@@ -31,7 +32,7 @@
     }

     dependencies {
-        classpath 'com.android.tools.build:gradle:3.0.1'
+        classpath 'com.android.tools.build:gradle:3.5.3'
         classpath 'com.github.dcendents:android-maven-gradle-plugin:1.5'
         classpath 'com.jfrog.bintray.gradle:gradle-bintray-plugin:1.7.3'
     }
===================================================================
--- origin_platforms/android/app/build.gradle	null
+++ platforms/android/app/build.gradle	null
@@ -21,6 +21,7 @@

 buildscript {
     repositories {
+        google()
         mavenCentral()
         maven {
             url "https://maven.google.com"
@@ -29,7 +30,7 @@
     }

     dependencies {
-        classpath 'com.android.tools.build:gradle:3.0.1'
+        classpath 'com.android.tools.build:gradle:3.5.3'
     }
 }

@@ -145,31 +146,43 @@
     return computeBuildTargetName(false)
 }

-task cdvPrintProps << {
-    println('cdvCompileSdkVersion=' + cdvCompileSdkVersion)
-    println('cdvBuildToolsVersion=' + cdvBuildToolsVersion)
-    println('cdvVersionCode=' + cdvVersionCode)
-    println('cdvVersionCodeForceAbiDigit=' + cdvVersionCodeForceAbiDigit)
-    println('cdvMinSdkVersion=' + cdvMinSdkVersion)
-    println('cdvBuildMultipleApks=' + cdvBuildMultipleApks)
-    println('cdvReleaseSigningPropertiesFile=' + cdvReleaseSigningPropertiesFile)
-    println('cdvDebugSigningPropertiesFile=' + cdvDebugSigningPropertiesFile)
-    println('cdvBuildArch=' + cdvBuildArch)
-    println('computedVersionCode=' + android.defaultConfig.versionCode)
-    android.productFlavors.each { flavor ->
-        println('computed' + flavor.name.capitalize() + 'VersionCode=' + flavor.versionCode)
+task cdvPrintProps {
+    doLast {
+        println('cdvCompileSdkVersion=' + cdvCompileSdkVersion)
+        println('cdvBuildToolsVersion=' + cdvBuildToolsVersion)
+        println('cdvVersionCode=' + cdvVersionCode)
+        println('cdvVersionCodeForceAbiDigit=' + cdvVersionCodeForceAbiDigit)
+        println('cdvMinSdkVersion=' + cdvMinSdkVersion)
+        println('cdvBuildMultipleApks=' + cdvBuildMultipleApks)
+        println('cdvReleaseSigningPropertiesFile=' + cdvReleaseSigningPropertiesFile)
+        println('cdvDebugSigningPropertiesFile=' + cdvDebugSigningPropertiesFile)
+        println('cdvBuildArch=' + cdvBuildArch)
+        println('computedVersionCode=' + android.defaultConfig.versionCode)
+        android.productFlavors.each { flavor ->
+            println('computed' + flavor.name.capitalize() + 'VersionCode=' + flavor.versionCode)
+        }
     }
 }

 android {

+    useLibrary 'org.apache.http.legacy'
+
     defaultConfig {
-        versionCode cdvVersionCode ?: new BigInteger("" + privateHelpers.extractIntFromManifest("versionCode"))
         applicationId privateHelpers.extractStringFromManifest("package")

         if (cdvMinSdkVersion != null) {
             minSdkVersion cdvMinSdkVersion
+        } else {
+            minSdkVersion 21
         }
+        targetSdkVersion 28
+        versionCode 10000
+        versionName "1.0.0"
+    }
+
+    sourceSets {
+        main.java.srcDirs += '../../../platform_src/android/app/src/main/java'
     }

     lintOptions {
@@ -263,6 +276,7 @@

 dependencies {
     implementation fileTree(dir: 'libs', include: '*.jar')
+    implementation 'io.jsonwebtoken:jjwt:0.9.0'
+    implementation 'com.android.support:appcompat-v7:23.1.1'
     // SUB-PROJECT DEPENDENCIES START
     implementation(project(path: ":CordovaLib"))
     // SUB-PROJECT DEPENDENCIES END
}

===================================================================
--- origin_platforms/android/build.gradle	null
+++ platforms/android/build.gradle	null
@@ -20,6 +20,7 @@

 buildscript {
     repositories {
+        google()
         maven {
             url "https://maven.google.com"
         }
@@ -29,7 +30,7 @@

         // NOTE: Do not place your application dependencies here; they belong
         // in the individual module build.gradle files
-        classpath 'com.android.tools.build:gradle:3.0.1'
+        classpath 'com.android.tools.build:gradle:3.5.3'
     }
 }

@@ -43,7 +44,7 @@
     //This replaces project.properties w.r.t. build settings
     project.ext {
       defaultBuildToolsVersion="27.0.1" //String
-      defaultMinSdkVersion=19 //Integer - Minimum requirement is Android 4.4
+      defaultMinSdkVersion=21 //Integer - Minimum requirement is Android 5
       defaultTargetSdkVersion=27 //Integer - We ALWAYS target the latest by default
       defaultCompileSdkVersion=27 //Integer - We ALWAYS compile with the latest by default
     }
