===================================================================
--- origin_platforms/android/cordova/lib/builders/GradleBuilder.js	null
+++ platforms/android/cordova/lib/builders/GradleBuilder.js	null
@@ -207,7 +207,7 @@
                 throw new CordovaError('Unsupported system library (does not work with gradle): ' + p);
             }
         }
-        depsList += '    compile "' + mavenRef + '"\n';
+        depsList += '    implementation "' + mavenRef + '"\n';
     });

     // This code is dangerous and actually writes gradle declarations directly into the build.gradle
@@ -248,7 +248,7 @@
         // For some reason, using ^ and $ don't work.  This does the job, though.
         var distributionUrlRegex = /distributionUrl.*zip/;
         /* jshint -W069 */
-        var distributionUrl = process.env['CORDOVA_ANDROID_GRADLE_DISTRIBUTION_URL'] || 'https\\://services.gradle.org/distributions/gradle-4.1-all.zip';
+        var distributionUrl = process.env['CORDOVA_ANDROID_GRADLE_DISTRIBUTION_URL'] || 'https\\://services.gradle.org/distributions/gradle-5.4.1-all.zip';
         /* jshint +W069 */
         var gradleWrapperPropertiesPath = path.join(self.root, 'gradle', 'wrapper', 'gradle-wrapper.properties');
         shell.chmod('u+w', gradleWrapperPropertiesPath);
===================================================================
--- origin_platforms/android/cordova/lib/builders/StudioBuilder.js	null
+++ platforms/android/cordova/lib/builders/StudioBuilder.js	null
@@ -198,7 +198,7 @@
                 throw new CordovaError('Unsupported system library (does not work with gradle): ' + p);
             }
         }
-        depsList += '    compile "' + mavenRef + '"\n';
+        depsList += '    implementation "' + mavenRef + '"\n';
     });

     buildGradle = buildGradle.replace(/(SUB-PROJECT DEPENDENCIES START)[\s\S]*(\/\/ SUB-PROJECT DEPENDENCIES END)/, '$1\n' + depsList + '    $2');
@@ -224,7 +224,7 @@
             // If it's not set, do nothing, assuming that we're using a future version of gradle that we don't want to mess with.
             // For some reason, using ^ and $ don't work.  This does the job, though.
             var distributionUrlRegex = /distributionUrl.*zip/;
-            var distributionUrl = process.env['CORDOVA_ANDROID_GRADLE_DISTRIBUTION_URL'] || 'https\\://services.gradle.org/distributions/gradle-4.1-all.zip';
+            var distributionUrl = process.env['CORDOVA_ANDROID_GRADLE_DISTRIBUTION_URL'] || 'https\\://services.gradle.org/distributions/gradle-5.4.1-all.zip';
             var gradleWrapperPropertiesPath = path.join(self.root, 'gradle', 'wrapper', 'gradle-wrapper.properties');
             shell.chmod('u+w', gradleWrapperPropertiesPath);
             shell.sed('-i', distributionUrlRegex, 'distributionUrl=' + distributionUrl, gradleWrapperPropertiesPath);
