--- origin_plugins/cordova-plugin-device-motion/www/Acceleration.js	null
+++ plugins/cordova-plugin-device-motion/www/Acceleration.js	null
@@ -19,6 +19,16 @@
  *
 */
 
+/**
+ * Contains data captured at a specific point in time.
+ *
+ * @exports Acceleration
+ * @param {Number} x      				  The amount of acceleration on the x-axis.
+ * @param {Number} y      			   	The amount of acceleration on the y-axis.
+ * @param {Number} z      			   	The amount of acceleration on the z-axis.
+ * @param {DOMTimeStamp} timestamp  The creation timestamp in milliseconds.
+ */
+
 var Acceleration = function(x, y, z, timestamp) {
     this.x = x;
     this.y = y;
