diff -ruNb origin_plugins/cordova-plugin-device/types/index.d.ts plugins/cordova-plugin-device/types/index.d.ts
--- origin_plugins/cordova-plugin-device/types/index.d.ts	null
+++ plugins/cordova-plugin-device/types/index.d.ts	null
@@ -7,8 +7,17 @@
 /**
  * This plugin defines a global device object, which describes the device's hardware and software.
  * Although the object is in the global scope, it is not available until after the deviceready event.
+ * <br><br>
+ * Please use 'Device' as the plugin name in the manifest.json if you want to use
+ * this facility.
+ * <br><br>
+ * Usage:
+ * <br>
+ * device.platform;
  */
-interface Device {
+
+declare namespace CordovaDevicePlugin {
+    interface Device {
     /** Get the version of Cordova running on the device. */
     cordova: string;
     /** Indicates that Cordova initialize successfully. */
@@ -30,6 +39,7 @@
     isVirtual: boolean;
     /** Get the device hardware serial number. */
     serial: string;
+    }
 }
 
-declare var device: Device;
\ No newline at end of file
+declare var device: CordovaDevicePlugin.Device;
\ No newline at end of file
diff -ruNb origin_plugins/cordova-plugin-device/www/device.js plugins/cordova-plugin-device/www/device.js
--- origin_plugins/cordova-plugin-device/www/device.js	null
+++ plugins/cordova-plugin-device/www/device.js	null
@@ -33,6 +33,14 @@
  * This represents the mobile device, and provides properties for inspecting the model, version, UUID of the
  * phone, etc.
  * @constructor
+ * @property {boolean} cordova    - Get the version of Cordova running on the device.
+ * @property {string}  model      - The device.model returns the name of the device's model or product.
+ * @property {string}  platform   - Get the device's operating system name.
+ * @property {string} uuid       - Get the device's Universally Unique Identifier.
+ * @property {string} version    - Get the operating system version.
+ * @property {string} manufacturer - Get the device's manufacturer.
+ * @property {boolean} isVirtual - whether the device is running on a simulator.
+ * @property {string} serial     - Get the device hardware serial number.
  */
 function Device () {
     this.available = false;
