diff -ruNb origin_plugins/cordova-plugin-device-motion/www/Acceleration.js plugins/cordova-plugin-device-motion/www/Acceleration.js
--- origin_plugins/cordova-plugin-device-motion/www/Acceleration.js	null
+++ plugins/cordova-plugin-device-motion/www/Acceleration.js	null
@@ -19,6 +19,10 @@
  *
 */

+/**
+* @exports Acceleration
+*/
+
 var Acceleration = function(x, y, z, timestamp) {
     this.x = x;
     this.y = y;
diff -ruNb origin_plugins/cordova-plugin-device-motion/www/accelerometer.js plugins/cordova-plugin-device-motion/www/accelerometer.js
--- origin_plugins/cordova-plugin-device-motion/www/accelerometer.js	null
+++ plugins/cordova-plugin-device-motion/www/accelerometer.js	null
@@ -19,10 +19,14 @@
  *
 */

-/**
+/*
  * This class provides access to device accelerometer data.
- * @constructor
  */
+
+/**
+* @exports Accelerometer
+*/
+
 var argscheck = require('cordova/argscheck'),
     utils = require("cordova/utils"),
     exec = require("cordova/exec"),
