diff -ruNb origin_plugins/cordova-plugin-network-information/types/index.d.ts plugins/cordova-plugin-network-information/types/index.d.ts
--- origin_plugins/cordova-plugin-network-information/types/index.d.ts	null
+++ plugins/cordova-plugin-network-information/types/index.d.ts	null
@@ -6,33 +6,24 @@
 // Copyright (c) Microsoft Open Technologies Inc
 // Licensed under the MIT license
 
-interface Navigator {
-    /**
-     * This plugin provides an implementation of an old version of the Network Information API.
-     * It provides information about the device's cellular and wifi connection, and whether the device has an internet connection.
-     */
-    connection: Connection;
-    // see https://github.com/apache/cordova-plugin-network-information/blob/dev/doc/index.md#api-change
-    // for 
-    network: {
-        /**
+/**
          * This plugin provides an implementation of an old version of the Network Information API.
          * It provides information about the device's cellular and wifi connection, and whether the device has an internet connection.
+ * <br><br>
+ * Please use 'NetworkStatus' as the plugin name in the manifest.json if you want to use
+ * this facility.                                                   
+ * <br><br>                                                         
+ * Usage:                                                           
+ * <br>                                                             
+ * navigator.connection.type;
          */
-        connection: Connection
-    }
-}
-
-interface Document {
-    addEventListener(type: "online", connectionStateCallback: () => any, useCapture?: boolean): void;
-    addEventListener(type: "offline", connectionStateCallback: () => any, useCapture?: boolean): void;
-}
 
-/**
+declare namespace CordovaNetworkInfoPlugin {
+    /**
  * The connection object, exposed via navigator.connection, provides information
  * about the device's cellular and wifi connection.
  */
-interface Connection {
+    interface Connection {
     /**
      * This property offers a fast way to determine the device's network connection state, and type of connection.
      * One of:
@@ -48,9 +39,9 @@
     type: string;
     addEventListener(type: string, listener: EventListenerOrEventListenerObject, useCapture?: boolean): void;
     removeEventListener(type: string, listener: EventListenerOrEventListenerObject, useCapture?: boolean): void;
-}
+    }
 
-declare var Connection: {
+    let Connection: {
     UNKNOWN: string;
     ETHERNET: string;
     WIFI: string;
@@ -59,4 +50,19 @@
     CELL_4G: string;
     CELL: string;
     NONE: string;
+    }
+}
+
+interface Navigator {
+    connection: CordovaNetworkInfoPlugin.Connection;
+    // see https://github.com/apache/cordova-plugin-network-information/blob/dev/doc/index.md#api-change
+    // for 
+    network: {
+        connection: CordovaNetworkInfoPlugin.Connection
+    }
+}
+
+interface Document {
+    addEventListener(type: "online", connectionStateCallback: () => any, useCapture?: boolean): void;
+    addEventListener(type: "offline", connectionStateCallback: () => any, useCapture?: boolean): void;
 }
\ No newline at end of file
diff -ruNb origin_plugins/cordova-plugin-network-information/www/Connection.js plugins/cordova-plugin-network-information/www/Connection.js
--- origin_plugins/cordova-plugin-network-information/www/Connection.js	null
+++ plugins/cordova-plugin-network-information/www/Connection.js	null
@@ -19,8 +19,14 @@
  *
 */
 
+
 /**
+ * @module Connection
+ *
+ * @description
  * Network status
+ *
+ * @enum {number}
  */
 module.exports = {
     UNKNOWN: 'unknown',
diff -ruNb origin_plugins/cordova-plugin-network-information/www/network.js plugins/cordova-plugin-network-information/www/network.js
--- origin_plugins/cordova-plugin-network-information/www/network.js	null
+++ plugins/cordova-plugin-network-information/www/network.js	null
@@ -88,4 +88,9 @@
     });
 });
 
+/**
+ * @module Connection
+ * @example
+ * var connectionStatus = navigator.connection.type;
+ */
 module.exports = me;
