diff -ruNb origin_plugins/cordova-plugin-vibration/www/vibration.js plugins/cordova-plugin-vibration/www/vibration.js
--- origin_plugins/cordova-plugin-vibration/www/vibration.js	null
+++ plugins/cordova-plugin-vibration/www/vibration.js	null
@@ -25,6 +25,9 @@
  * Provides access to the vibration mechanism on the device.
  */

+/**
+ * @exports Vibration
+ */
 module.exports = {

     /**
@@ -33,7 +36,7 @@
      * @param {Integer} param       The number of milliseconds to vibrate (if 0, cancels vibration)
      *
      *
-     * @param {Array of Integer} param    Pattern with which to vibrate the device.
+     * @param {number[]} param    Pattern with which to vibrate the device.
      *                                      Pass in an array of integers that
      *                                      are the durations for which to
      *                                      turn on or off the vibrator in
