===================================================================
--- origin_platforms/android/CordovaLib/build.gradle	null
+++ platforms/android/CordovaLib/build.gradle	null
@@ -24,6 +24,7 @@
 
 buildscript {
     repositories {
+        google()
         maven {
             url "https://maven.google.com"
         }
@@ -31,7 +32,7 @@
     }
 
     dependencies {
-        classpath 'com.android.tools.build:gradle:3.0.1'
+        classpath 'com.android.tools.build:gradle:3.2.1'
         classpath 'com.github.dcendents:android-maven-gradle-plugin:1.5'
         classpath 'com.jfrog.bintray.gradle:gradle-bintray-plugin:1.7.3'
     }
===================================================================
--- origin_platforms/android/app/build.gradle	null
+++ platforms/android/app/build.gradle	null
@@ -21,6 +21,7 @@
 
 buildscript {
     repositories {
+        google()
         mavenCentral()
         maven {
             url "https://maven.google.com"
@@ -29,7 +30,7 @@
     }
 
     dependencies {
-        classpath 'com.android.tools.build:gradle:3.0.1'
+        classpath 'com.android.tools.build:gradle:3.2.1'
     }
 }
 
@@ -170,9 +171,15 @@
 
         if (cdvMinSdkVersion != null) {
             minSdkVersion cdvMinSdkVersion
+        } else {
+            minSdkVersion 19
         }
     }
 
+    sourceSets {
+        main.java.srcDirs += '../../../platform_src/android/app/src/main/java'
+    }
+
     lintOptions {
       abortOnError false;
     }
@@ -266,7 +273,7 @@
     implementation fileTree(dir: 'libs', include: '*.jar')
     // SUB-PROJECT DEPENDENCIES START
     implementation(project(path: ":CordovaLib"))
-    compile "com.android.support:support-v4:27.+"
+    implementation 'com.android.support:support-v4:27.+'
     // SUB-PROJECT DEPENDENCIES END
 }
 
===================================================================
--- origin_platforms/android/elastos-trinity-plugins-qrscanner/runtime-qrscanner.gradle	null
+++ platforms/android/elastos-trinity-plugins-qrscanner/runtime-qrscanner.gradle	null
@@ -3,8 +3,8 @@
 }
 
 dependencies {
-    compile 'com.journeyapps:zxing-android-embedded:3.3.0'
-    compile 'com.android.support:appcompat-v7:23.1.0'
+    implementation 'com.journeyapps:zxing-android-embedded:3.3.0'
+    implementation 'com.android.support:appcompat-v7:23.1.0'
 }
 
 android {
===================================================================
--- origin_platforms/android/build.gradle	null
+++ platforms/android/build.gradle	null
@@ -20,6 +20,7 @@
 
 buildscript {
     repositories {
+        google()
         maven {
             url "https://maven.google.com"
         }
@@ -29,7 +30,7 @@
 
         // NOTE: Do not place your application dependencies here; they belong
         // in the individual module build.gradle files
-        classpath 'com.android.tools.build:gradle:3.0.1'
+        classpath 'com.android.tools.build:gradle:3.2.1'
     }
 }
 
